### DISCUSSION TITLE
Sugestão de padrão

### DISCUSSION THREAD

**Discussion by barbier** [frontendbr/forum#213] [Perguntas e Respostas]
Olá!
Ontem eu publiquei o que considero o meu primeiro projeto Open-Source de verdade, o [Mana Calc](https://github.com/barbier/mana-calc).
Como vocês podem ver no arquivo app.js, está com muita repetição de código e foi um porre ter que digitar aquilo tudo. Apesar da hora, antes de dormir eu pensei em refatorar usando um array. No caso, eu usaria um array para armazenar os valores dos inputs e faria os cálculos através dos índices.
Me sugerem algo diferente? Existe algum padrão específico para esse tipo de aplicação?

Obrigado =)

**Comment by Adejair** [comment]
Você poderia usar um regex para poder selecionar os itens, e dentro da iteração você poderia fazer os cálculos, com o valor do índice + 1 (pois começa em 0)

[http://stackoverflow.com/questions/3086554/find-all-elements-based-on-ids-using-regex-on-jquery-selector](http://stackoverflow.com/questions/3086554/find-all-elements-based-on-ids-using-regex-on-jquery-selector)

Algo assim, 

``` javascript
$('yourRegex').each((v, i) => {
    total += parseInt($(this).val()) * (i + 1);
});
```

**Comment by barbier** [comment]
A princípio eu pensei em pegar a coleção de inputs, algo como `$("input[type='number']")` e inserir num array, mas vou olhar com calma esse link do SO.
Obrigado! =)

**Comment by renatogalvones** [comment]
@barbier Sugestão: Acredito que seria uma boa ao invés do pessoal ter q baixar seu projeto pra usar, seria vc usar o ghpages pra hostear teu projeto. =]

**Comment by barbier** [comment]
@renatogalvones, será feito nos próximos dias, meu aniversário é amanhã então eu só terei como mexer nisso semana que vem, já me cobraram lá no projeto =P
Eu queria ter lançado direto como uma gh-page, mas como eu não tinha entendido algumas coisas e queria lançar logo, achei melhor deixar assim uns dias e me entender com a page depois.

**Comment by renatogalvones** [comment]
@barbier Cria uma branch com o nome **gh-pages** e depois acessa como http://barbier.github.io/mana-calc

**Comment by barbier** [comment]
@renatogalvones, WOW!
Isso foi infinitamente mais fácil do que eu achei que seria!
Achei que eu teria que fazer alguns ajustes e subir a branch a partir da minha máquina, mas já está online!

**Comment by rdiego26** [comment]
@barbier 
Se quiser entender ( não tem nada de segredo.rs ) a fundo como funciona, indico a leitura [aqui](http://blog.da2k.com.br/2015/02/05/segredos-do-github-hospedando-seu-site-no-github/) .

**Comment by barbier** [comment]
Diego, vou olhar com calma depois, obrigado!

**Comment by renatogalvones** [comment]
@barbier Nozes! Sempre achei q era um bicho de 7 cabeças até um brother me passar esse tutorial do @fdaciuk

**Comment by fdaciuk** [comment]
@barbier acho que usando array é a melhor opção. De 48 linhas vai ficar com, no máximo, umas 10 =)
Para os itens do DOM, você pode usar `Array.prototype.map.apply`, e fazer o array-like se comportar conseguir usar os mesmos métodos de array xD

Para os cálculos, dá pra usar `reduce` e boa! =)
